### PR TITLE
[1.10] Add DifficultyChangeEvent

### PR DESCRIPTION
**Pull request by 0f-0b**
See #3088

### CODE REVIEW

File: src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{
+    private final EnumDifficulty oldDifficulty;
+    private EnumDifficulty newDifficulty;
+    public DifficultyChangeEvent(EnumDifficulty oldDifficulty, EnumDifficulty newDifficulty)
+    {
+        this.oldDifficulty = oldDifficulty;
+        this.newDifficulty = newDifficulty;
+    }
+
+    public EnumDifficulty getOldDifficulty() { return oldDifficulty; }
```
Comment (liach) [inline src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java:51]: This does not match the requirement of the code style.

File: src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    public EnumDifficulty getNewDifficulty() { return newDifficulty; }
```
Comment (liach) [inline src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java:53]: This does not match the requirement of the code style.

**Comment by williewillus** [comment]
What is the use case for this?
Not everything needs to be an event, especially something this small (it'll just pollute the event set with a bunch of tiny events like this)
You can just wrap things that need to be difficulty-aware in an if statement in a place that ticks

**Comment by liach** [comment]
@williewillus It allows mods to prevent the change in difficulty.

Edit: @sjx233 I think you can speak Mandarin, can't you? 呵呵

File: src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java (hunk 1 above)
Comment (sjx233) [inline src/main/java/net/minecraftforge/event/DifficultyChangeEvent.java]: How about
```java
    public EnumDifficulty getOldDifficulty()
    {
        return oldDifficulty;
    }
```

**Comment by BlayTheNinth** [comment]
Preventing the change in difficulty could be done in different places for more responsive cancelling, e.g. listening on the ServerCommandEvent to display an error message instead of executing the /difficulty command, or disabling the difficulty button in GuiScreenEvent.InitGui.

**Comment by wshadow** [comment]
@williewillus: polling for the current value in every tick just asks for performance problems in my opinion.

Especially for the difficulty value (which will not change very often in most games) polling seems like a lot of wasted processing time.

As for your question about the usecase:

> My idea was to add the current difficulty as a 'task' to the BetterQuesting mod. With this I'd be able to react to the (currently) set difficulty level and offer different quests/rewards based on that.

(see also #3088 )

**Comment by mezz** [comment]
Please update this to target 1.11, new events are no longer being merged into 1.10.

**Comment by 0f-0b** [comment]
See #3525

**Pull request closed without merging**